crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -3,74 +3,172 @@ use containerd_shim_wasm::container::{
     Engine, Instance, PathResolve, RuntimeContext, Stdio, WasiEntrypoint,
 };
 use wasi_common::I32Exit;
-use wasmtime::{Linker, Module, Store};
+use wasmtime::component::Component;
+use wasmtime::{Module, Store};
+use wasmtime_wasi::preview2::{self as wasi_preview2, Table};
 use wasmtime_wasi::{Dir, WasiCtxBuilder};
 
 pub type WasmtimeInstance = Instance<WasmtimeEngine>;
 
-#[derive(Clone, Default)]
+#[derive(Clone)]
 pub struct WasmtimeEngine {
     engine: wasmtime::Engine,
 }
 
+impl Default for WasmtimeEngine {
+    fn default() -> Self {
+        let mut config = wasmtime::Config::new();
+        config.wasm_component_model(true);
+        Self {
+            engine: wasmtime::Engine::new(&config)
+                .context("failed to create wasmtime engine")
+                .unwrap(),
+        }
+    }
+}
+
+pub struct Data {
+    wasi: wasi_preview2::WasiCtx,
+    table: Table,
+}
+
+impl wasmtime_wasi::preview2::WasiView for Data {
```

Comment:
I'm surprised `wasmtime_wasi` doesn't provide an off the shelf way of doing this

Reply:
Sadly, true.